File: system-tests-fixtures/api/TestApi/android/lib/src/main/java/com/TestApi/ern/api/WalmartItemEvents.java
/*
* Copyright 2017 WalmartLabs
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* http://www.apache.org/licenses/LICENSE-2.0
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/

package com.TestApi.ern.api;

import android.support.annotation.NonNull;

import com.walmartlabs.electrode.reactnative.bridge.ElectrodeBridgeEventListener;
import com.walmartlabs.electrode.reactnative.bridge.ElectrodeBridgeEvent;
import com.walmartlabs.electrode.reactnative.bridge.ElectrodeBridgeHolder;
import com.walmartlabs.electrode.reactnative.bridge.EventListenerProcessor;
import com.walmartlabs.electrode.reactnative.bridge.EventProcessor;
import com.walmartlabs.electrode.reactnative.bridge.None;
import java.util.*;
import java.util.UUID;

    import com.TestApi.ern.model.Item;

final class WalmartItemEvents implements WalmartItemApi.Events {
    WalmartItemEvents() {}

    @Override
    public UUID addItemAddedEventListener(@NonNull final ElectrodeBridgeEventListener<String> eventListener) {
        return new EventListenerProcessor<>(EVENT_ITEM_ADDED, String.class, eventListener).execute();
    }

                @Override
                public ElectrodeBridgeEventListener<ElectrodeBridgeEvent> removeItemAddedEventListener(@NonNull final UUID uuid) {
                    return ElectrodeBridgeHolder.removeEventListener(uuid);
                }

    //------------------------------------------------------------------------------------------------------------------------------------

    @Override
    public void emitItemAdded(String itemId) {
        new EventProcessor<>(EVENT_ITEM_ADDED, itemId).execute();
    }
}

File: system-tests/fixtures/api/test-api/javascript/src/api/WalmartItemEvents.js
// @flow

export default class WalmartItemEvents {
  _bridge: Object;

  constructor(bridge: Object) {
    this._bridge = bridge;
  }

  addItemAddedEventListener(eventListener: Function): string {
    return this._bridge.registerEventListener(
      'com.test.ern.api.event.itemAdded',
      eventListener,
    );
  }

  removeItemAddedEventListener(uuid: string): any {
    return this._bridge.removeEventListener(uuid);
  }

  emitItemAdded(itemId: string): void {
    return this._bridge.emitEvent('com.test.ern.api.event.itemAdded', {
      data: itemId,
    });
  }
}
